hail/python/hailtop/batch_client/aioclient.py

```diff
@@ -56,17 +56,14 @@ def total_duration(job_status):
             return None
 
         durations = job_status['duration']
-        durations = [durations[task] for task in ['setup', 'main', 'cleanup'] if task in durations]
 
-        i = 0
-        duration = 0
-        while i < len(durations):
-            d = durations[i]
-            if d is None:
-                return None
-            duration += d
-            i += 1
-        return duration
+        setup_duration = durations.get('setup', 0)
+        main_duration = durations.get('main', 0)
+        cleanup_duration = durations.get('cleanup', 0)
+        if setup_duration is None or main_duration is None or cleanup_duration is None:
```

Comment:
Will this happen if you're setting the defaults to 0?

Reply:
Yes:
```
>>> x = {'a': 3, 'b': None}
>>> print(x.get('a', 0))
3
>>> print(x.get('b', 0))
None
>>> print(x.get('c', 0))
0
```